boards/arduino-mega2560/include/arduino_pinmap.h

```diff
@@ -31,93 +32,78 @@ extern "C" {
-#define ARDUINO_PIN_89          GPIO_PIN(PORT_K, 0)
-#define ARDUINO_PIN_90          GPIO_PIN(PORT_F, 7)
-#define ARDUINO_PIN_91          GPIO_PIN(PORT_F, 6)
-#define ARDUINO_PIN_92          GPIO_PIN(PORT_F, 5)
-#define ARDUINO_PIN_93          GPIO_PIN(PORT_F, 4)
-#define ARDUINO_PIN_94          GPIO_PIN(PORT_F, 3)
-#define ARDUINO_PIN_95          GPIO_PIN(PORT_F, 2)
-#define ARDUINO_PIN_96          GPIO_PIN(PORT_F, 1)
-#define ARDUINO_PIN_97          GPIO_PIN(PORT_F, 0)
+#define ARDUINO_PIN_0		GPIO_PIN(PORT_E, 0)
+#define ARDUINO_PIN_1		GPIO_PIN(PORT_E, 1)
+#define ARDUINO_PIN_2		GPIO_PIN(PORT_E, 4)
+#define ARDUINO_PIN_3		GPIO_PIN(PORT_E, 5)
+#define ARDUINO_PIN_4		GPIO_PIN(PORT_G, 5)
+#define ARDUINO_PIN_5		GPIO_PIN(PORT_E, 3)
+#define ARDUINO_PIN_6		GPIO_PIN(PORT_H, 3)
+#define ARDUINO_PIN_7		GPIO_PIN(PORT_H, 4)
+#define ARDUINO_PIN_8		GPIO_PIN(PORT_H, 5)
+#define ARDUINO_PIN_9		GPIO_PIN(PORT_H, 6)
+#define ARDUINO_PIN_10		GPIO_PIN(PORT_B, 4)
+#define ARDUINO_PIN_11		GPIO_PIN(PORT_B, 5)
+#define ARDUINO_PIN_12		GPIO_PIN(PORT_B, 6)
+#define ARDUINO_PIN_13		GPIO_PIN(PORT_B, 7)
+#define ARDUINO_PIN_14		GPIO_PIN(PORT_J, 1)
+#define ARDUINO_PIN_15		GPIO_PIN(PORT_J, 0)
+#define ARDUINO_PIN_16		GPIO_PIN(PORT_H, 1)
+#define ARDUINO_PIN_17		GPIO_PIN(PORT_H, 0)
+#define ARDUINO_PIN_18		GPIO_PIN(PORT_D, 3)
+#define ARDUINO_PIN_19		GPIO_PIN(PORT_D, 2)
+#define ARDUINO_PIN_20		GPIO_PIN(PORT_D, 1)
+#define ARDUINO_PIN_21		GPIO_PIN(PORT_D, 0)
+#define ARDUINO_PIN_22		GPIO_PIN(PORT_A, 0)
+#define ARDUINO_PIN_23		GPIO_PIN(PORT_A, 1)
+#define ARDUINO_PIN_24		GPIO_PIN(PORT_A, 2)
+#define ARDUINO_PIN_25		GPIO_PIN(PORT_A, 3)
+#define ARDUINO_PIN_26		GPIO_PIN(PORT_A, 4)
+#define ARDUINO_PIN_27		GPIO_PIN(PORT_A, 5)
+#define ARDUINO_PIN_28		GPIO_PIN(PORT_A, 6)
+#define ARDUINO_PIN_29		GPIO_PIN(PORT_A, 7)
+#define ARDUINO_PIN_30		GPIO_PIN(PORT_C, 7)
+#define ARDUINO_PIN_31		GPIO_PIN(PORT_C, 6)
+#define ARDUINO_PIN_32		GPIO_PIN(PORT_C, 5)
+#define ARDUINO_PIN_33		GPIO_PIN(PORT_C, 4)
+#define ARDUINO_PIN_34		GPIO_PIN(PORT_C, 3)
+#define ARDUINO_PIN_35		GPIO_PIN(PORT_C, 2)
+#define ARDUINO_PIN_36		GPIO_PIN(PORT_C, 1)
+#define ARDUINO_PIN_37		GPIO_PIN(PORT_E, 0)
+#define ARDUINO_PIN_38		GPIO_PIN(PORT_D, 7)
+#define ARDUINO_PIN_39		GPIO_PIN(PORT_G, 2)
+#define ARDUINO_PIN_40		GPIO_PIN(PORT_G, 1)
+#define ARDUINO_PIN_41		GPIO_PIN(PORT_G, 0)
+#define ARDUINO_PIN_42		GPIO_PIN(PORT_L, 7)
+#define ARDUINO_PIN_43		GPIO_PIN(PORT_L, 6)
+#define ARDUINO_PIN_44		GPIO_PIN(PORT_L, 5)
+#define ARDUINO_PIN_45		GPIO_PIN(PORT_L, 4)
+#define ARDUINO_PIN_46		GPIO_PIN(PORT_L, 3)
+#define ARDUINO_PIN_47		GPIO_PIN(PORT_L, 2)
+#define ARDUINO_PIN_48		GPIO_PIN(PORT_L, 1)
+#define ARDUINO_PIN_49		GPIO_PIN(PORT_L, 0)
+#define ARDUINO_PIN_50		GPIO_PIN(PORT_B, 3)
+#define ARDUINO_PIN_51		GPIO_PIN(PORT_B, 2)
+#define ARDUINO_PIN_52		GPIO_PIN(PORT_B, 1)
+#define ARDUINO_PIN_53		GPIO_PIN(PORT_B, 0)
+#define ARDUINO_PIN_54		GPIO_PIN(PORT_F, 0)		//Analog pin 0
```

Comment:
I completely agree with this, the reson why I didn't do it was, that this is the way its done in the Arduino due's pin mapping. But ill be happy to change it since it makes alot more sense.